reportsizedeltas/reportsizedeltas.py

```diff
@@ -167,33 +169,53 @@ def report_size_deltas_from_workflow_artifacts(self) -> None:
             page_number += 1
             page_count = api_data["page_count"]
 
-    def report_exists(self, pr_number: int, pr_head_sha: str) -> bool:
+    def report_exists(self, pr_number: int, pr_head_sha: str = "") -> str | None:
         """Return whether a report has already been commented to the pull request thread for the latest workflow run.
 
-        Keyword arguments:
-        pr_number -- number of the pull request to check
-        pr_head_sha -- PR's head branch hash
+        If `pr_head_sha` is a blank str, then all thread comments are traversed. Additionally,
+        any report comment found will be deleted if it is not the first report comment found.
+        This is designed to support the action input `update-comment` when asserted.
+
+        If `pr_head_sha` is not a blank str, then thread comments are traversed
+        until a report comment that corresponds to the commit is found.
+        No comments are deleted in this scenario.
+
+        Arguments:
+          pr_number: number of the pull request to check
+          pr_head_sha: PR's head branch hash
+        Returns:
+          - A URL str to use for PATCHing the first applicable report comment.
+          - None if no applicable report comments exist.
         """
-        # Get the pull request's comments
+        comment_url: str | None = None
+        request_uri = f"repos/{self.repository_name}/issues/{pr_number}/comments"
         page_number = 1
         page_count = 1
         while page_number <= page_count:
+            # Get the pull request's comments
             api_data = self.api_request(
-                request="repos/" + self.repository_name + "/issues/" + str(pr_number) + "/comments",
+                request=request_uri,
                 page_number=page_number,
             )
 
             comments_data = api_data["json_data"]
             for comment_data in comments_data:
                 # Check if the comment is a report for the PR's head SHA
                 if comment_data["body"].startswith(self.report_key_beginning + pr_head_sha):
```

Comment:
The only improvement I can think of is prepending `self.report_key_beginning` with an HTML comment:
https://github.com/arduino/report-size-deltas/blob/13c94843d8f85b19839463bb7573da1cc4b34737/reportsizedeltas/reportsizedeltas.py#L68
where as using
```python
    report_key_beginning = "<!-- arduino/report-size-deltas -->\n**Memory usage change @ "
```
would better assure that this action doesn't attempt to alter some user's comment starting with `**Memory usage change @ `.